Generate logical property group predicates for CSSProperties.h/cpp

weinig opened the pull request:
#### 5015f9ffc28bc5615e5087d41401c6af25e9aa3d
<pre>
Generate logical property group predicates for CSSProperties.h/cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=287029">https://bugs.webkit.org/show_bug.cgi?id=287029</a>

Reviewed by Tim Nguyen.

Adds generation of predicate functions that check if a property is part
of a specific logical property group, allowing us to remove two hand
written functions and make the rest of the groups available for future
use as well.

CSSProperty::isSizingProperty remains hand written as it represents
multiple logical property groups.

* Source/WebCore/css/CSSProperty.cpp:
(WebCore::CSSProperty::isInsetProperty): Deleted.
(WebCore::CSSProperty::isMarginProperty): Deleted.
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/process-css-properties.py:
(StyleProperties.__init__):
(StyleProperties):
(GenerateCSSPropertyNames):

Canonical link: <a href="https://commits.webkit.org/289825@main">https://commits.webkit.org/289825@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8c07d96052cf9345ae26bc4562f0e7ddedab1690

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88084 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7600 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42519 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93032 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38834 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90135 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7981 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15778 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67991 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25722 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91086 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6111 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79712 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48354 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5885 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/34126 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37942 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76274 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35005 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94880 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15252 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/11206 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76848 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15507 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75568 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76088 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18712 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20476 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/18904 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8281 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15270 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/20571 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15012 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18457 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16794 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
Would be good if the .h was auto-generated, so someone adding a shorthand doesn't have an extra step.

nt1m reviewed (approved):
This is adding the extra overhead of adding the function in the header for all new logical shorthands, but I guess that's fine for now.

weinig committed:
Generate logical property group predicates for CSSProperties.h/cpp
https://bugs.webkit.org/show_bug.cgi?id=287029

Reviewed by Tim Nguyen.

Adds generation of predicate functions that check if a property is part
of a specific logical property group, allowing us to remove two hand
written functions and make the rest of the groups available for future
use as well.

CSSProperty::isSizingProperty remains hand written as it represents
multiple logical property groups.

* Source/WebCore/css/CSSProperty.cpp:
(WebCore::CSSProperty::isInsetProperty): Deleted.
(WebCore::CSSProperty::isMarginProperty): Deleted.
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/process-css-properties.py:
(StyleProperties.__init__):
(StyleProperties):
(GenerateCSSPropertyNames):

Canonical link: https://commits.webkit.org/289825@main
